sklearn/preprocessing/_encoders.py

```diff
@@ -0,0 +1,845 @@
+    >>> enc.categories_
+    [array(['Female', 'Male'], dtype=object), array([1, 2, 3], dtype=object)]
+    >>> enc.transform([['Female', 1], ['Male', 4]]).toarray()
+    array([[1., 0., 1., 0., 0.],
+           [0., 1., 0., 0., 0.]])
+    >>> enc.inverse_transform([[0, 1, 1, 0, 0], [0, 0, 0, 1, 0]])
+    array([['Male', 1],
+           [None, 2]], dtype=object)
+
+    See also
+    --------
+    sklearn.preprocessing.OrdinalEncoder : performs an ordinal (integer)
+      encoding of the categorical features.
+    sklearn.feature_extraction.DictVectorizer : performs a one-hot encoding of
+      dictionary items (also handles string-valued features).
+    sklearn.feature_extraction.FeatureHasher : performs an approximate one-hot
+      encoding of dictionary items or strings.
+    sklearn.preprocessing.LabelBinarizer : binarizes labels in a one-vs-all
+      fashion.
+    sklearn.preprocessing.MultiLabelBinarizer : transforms between iterable of
+      iterables and a multilabel format, e.g. a (samples x classes) binary
+      matrix indicating the presence of a class label.
+    """
+
+    def __init__(self, n_values=None, categorical_features=None,
+                 categories=None, sparse=True, dtype=np.float64,
+                 handle_unknown='error'):
+        self._categories = self.categories = categories
+        self.sparse = sparse
+        self.dtype = dtype
+        self.handle_unknown = handle_unknown
+        self._n_values = self.n_values = n_values
+        self._categorical_features = categorical_features
+        self.categorical_features = categorical_features
+
+    # Deprecated attributes
+
+    @property
+    @deprecated("The 'active_features_' attribute was deprecated in version "
+                "0.20 and will be removed 0.22.")
+    def active_features_(self):
+        check_is_fitted(self, 'categories_')
+        return self._active_features_
+
+    @property
+    @deprecated("The 'feature_indices_' attribute was deprecated in version "
+                "0.20 and will be removed 0.22.")
+    def feature_indices_(self):
+        check_is_fitted(self, 'categories_')
+        return self._feature_indices_
+
+    @property
+    @deprecated("The 'n_values_' attribute was deprecated in version "
+                "0.20 and will be removed 0.22.")
+    def n_values_(self):
+        check_is_fitted(self, 'categories_')
+        return self._n_values_
+
+    def _handle_deprecations(self, X):
+
+        # user manually set the categories -> never legacy mode
+        if self._categories is not None:
+            self._legacy_mode = False
+
+        # categories not set -> infer if we need legacy mode or not
+        elif self._n_values is not None and self._n_values != 'auto':
+            msg = (
+                "Passing 'n_values' is deprecated in version 0.20 and will be "
+                "removed in 0.22. You can use the 'categories' keyword "
+                "instead. 'n_values=n' corresponds to 'categories=[range(n)]'."
```

Comment:
Yep, good catch for the `n_features`. So that would translate to `categories=[range(n)] * n_features`

> Should we support the case where each feature has the same categories more explicitly, or is it too weird?

That would be nice. In principle we could check the first element of the list, and if that is not an array/list, assume it is single list/array of categories that is used for all features.